File: recipes/jvm/java-recipes/micronaut-postgres-recipe/src/main/java/com/trendyol/stove/recipes/micronaut/infra/api/ProductController.java
package com.trendyol.stove.recipes.micronaut.infra.api;

import com.trendyol.stove.recipes.micronaut.application.ProductService;
import com.trendyol.stove.recipes.micronaut.domain.Product;
import io.micronaut.http.annotation.Body;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Post;
import io.micronaut.http.annotation.QueryValue;
import reactor.core.publisher.Mono;

@Controller("/products")
public class ProductController {

  private final ProductService productService;

  public ProductController(ProductService productService) {
    this.productService = productService;
  }

  @Get("/index")
  public String index(@QueryValue(defaultValue = "default") String keyword) {
    return "Hi from Stove framework with " + keyword;
  }

  @Post("/create")
  public Mono<Product> createProduct(@Body CreateProductRequest request) {
    return productService.createProduct(request.id(), request.name(), request.supplierId());
  }
}

File: examples/micronaut-example/src/main/kotlin/stove/micronaut/example/infrastructure/api/ProductController.kt
package stove.micronaut.example.infrastructure.api

import io.micronaut.http.annotation.*
import stove.micronaut.example.application.domain.Product
import stove.micronaut.example.application.services.ProductService
import stove.micronaut.example.infrastructure.api.model.request.CreateProductRequest

@Controller("/products")
class ProductController(
  private val productService: ProductService
) {
  @Get("/index")
  fun get(
    @QueryValue keyword: String = "default"
  ): String = "Hi from Stove framework with $keyword"

  @Post("/create")
  suspend fun createProduct(
    @Body request: CreateProductRequest
  ): Product = productService.createProduct(
    id = request.id,
    productName = request.name,
    supplierId = request.supplierId
  )
}
